app/Http/Controllers/Import/ImportMappingsController.php

```
@@ -28,15 +34,29 @@
     {
         $request->validated();
         Log::debug('Saving new import mapping', ['user_id' => Auth::id()]);
+
+        // Convert index-based mapping to header-based if needed
+        $columnMapping = $request->getColumnMapping();
+        $headers = $request->input('headers', []);
+
+        // If headers are provided and mapping is index-based, convert to header-based
+        if (! empty($headers) && $this->mappingService->isIndexBasedMapping($columnMapping)) {
+            $columnMapping = $this->mappingService->convertIndexMappingToHeaders($columnMapping, $headers);
```

PHPStan (2.1.15)

Parameter #2 $headers of method App\Services\ImportMappingService::convertIndexMappingToHeaders() expects array, mixed given.


(argument.type)